app/models/course/condition/achievement.rb

```diff
@@ -32,7 +32,7 @@ def self.dependent_class
   end
 
   def self.on_dependent_status_change(achievement)
-    return if achievement.previous_changes.empty?
+    return if achievement.changes.empty?
```

Comment:
@lowjoel Updated the test to consider all the different saving scenario that achievement and level will go through. E.g. creating, updating of the object.